src/main/java/org/kiwiproject/registry/eureka/client/EurekaParser.java

```diff
@@ -18,18 +20,24 @@
 
     @SuppressWarnings("unchecked")
     public static List<EurekaInstance> parseEurekaResponse(Map<String, Object> response) {
-        var applications = ((Map<String, Object>) response.get("applications")).get("application");
+        checkArgumentNotNull(response, "Map data of Eureka response must not be null");
+
+        var applications = (Map<String, Object>) response.get("applications");
+
+        checkState(nonNull(applications), "Eureka data must contain a key 'applications' that contains a Map");
+
+        var applicationMap = applications.get("application");
 
         var eurekaInstances = new ArrayList<EurekaInstance>();
 
-        if (nonNull(applications)) {
-            if (applications instanceof List<?>) {
-                for (Map<String, Object> application : (List<Map<String, Object>>) applications) {
+        if (nonNull(applicationMap)) {
+            if (applicationMap instanceof List<?>) {
```

Comment:
This is a bit confusing to ask if something called "applicationMap" is a List. Right?